packages/medusa/src/models/line-item.ts

```diff
@@ -18,14 +18,31 @@ import { LineItemTaxLine } from "./line-item-tax-line"
 import { Order } from "./order"
 import { ProductVariant } from "./product-variant"
 import { Swap } from "./swap"
-import { generateEntityId } from "../utils/generate-entity-id"
-import { FeatureFlagColumn } from "../utils/feature-flag-decorators"
+import { generateEntityId } from "../utils"
+import {
+  FeatureFlagClassDecorators,
+  FeatureFlagColumn,
+  FeatureFlagDecorators,
+} from "../utils/feature-flag-decorators"
 import TaxInclusivePricingFeatureFlag from "../loaders/feature-flags/tax-inclusive-pricing"
+import OrderEditingFeatureFlag from "../loaders/feature-flags/order-editing"
+import { OrderEdit } from "./order-edit"
 
 @Check(`"fulfilled_quantity" <= "quantity"`)
 @Check(`"shipped_quantity" <= "fulfilled_quantity"`)
 @Check(`"returned_quantity" <= "quantity"`)
 @Check(`"quantity" > 0`)
+@FeatureFlagClassDecorators(OrderEditingFeatureFlag.key, [
+  Index(
+    "unique_li_original_item_id_order_edit_id_order_id",
+    ["order_edit_id", "original_item_id", "order_id"],
```

Comment:
nit: `order_id` is redundant here as an OrderEdit cannot span multiple orders

Reply:
true, but on the same order edit we don't want to have to cloned line item targeting the same original line item and then be associated to the order.
For any reason, if we have a duplicate cloned line item, then upon the confirmation when the order_id will be assigned we should ensure that there is not duplicate.

wdyt?

Reply:
Just so that I am understanding correctly this is what you mean?

1. Order Placed 
2. OrderEdit created; all items from the order are cloned 
3. Randomly one of the original items are cloned independently of the order edit
4. OrderEdit is confirmed; with the above constraint we ensure that the randomly cloned item and the item from the OrderEdit isn't duplicated?

Reply:
not quite as if it is cloned out of the order edit it is not taken into account by the constraint.
I am thinking more of:
1. A user accidentally duplicate a cloned line from an order edit attached to an order in the DB. there it will throw a unique error
2. A user use our api and for some reason add some extra logic and by accident cloned some items upon confirmation (very low prob but safe guard as I have seen that users does not lack of imagination sometimes 😂 )
wdyt?

Reply:
In both of these cases I don't see how the `order_id` is also necessary; isn't it sufficient to just have the unique constraint across `order_edit_id` and `original_item_id`?

Reply:
No in fact you are right, just re done the schema in my head and indeed order id is not necessary.